util/dvsim/dvsim.py

```diff
@@ -566,6 +608,17 @@ def main():
     else:
         proj_root = get_proj_root()
 
+    # Check if jobs are dispatched to external compute machines. If yes,
+    # then the entire proj_root needs to be copied over to the scratch area
+    # which is accessible to those machines. This may be change in future.
+    if ((os.environ.get('DVSIM_REMOTE_DISPATCH') is not None) and
+        (args.scratch_root != DEFAULT_SCRATCH_ROOT)):
```

Comment:
Nit: I'm not keen on the second check here, because the user might theoretically have explicitly set the scratch root to be `DEFAULT_SCRATCH_ROOT` and this would trigger "by accident". Maybe `resolve_scratch_root` needs to return a pair where one field says whether the user overrode the default? Then you could check that field here.

Also, you'll find that flake8 moans about this line because it lines up with the next one. One work-around is to add another level of parens around the whole `if` check, which indents this line by one more space.

Reply:
I think I don't understand this comment. I *thought* this check was supposed to be: "If DVSIM_REMOTE_DISPATCH is set and the user has overridden scratch root, copy the repository". For the second part ("user has overridden scratch root"), I'm suggesting that you should explicitly check whether the user passed `--scratch-root`, rather than checking whether it's equal to the default value.

Reply:
If the user set `--scratch-root` but that is same as the tool's default scratch space, then copying is not needed. At Google, we are not passing `--scratch-root`. We are relying on `SCRATCH_ROOT` env var which gets set for everyone as a part of our opentitan modulefile.

Reply:
Ah, I see that you've changed this code to make it more explicit. Looks good to me!